ansible/roles/texlive/tasks/main.yml

```diff
@@ -0,0 +1,63 @@
+- name: texlive | prereq
+  yum: name={{ item }} state=latest
+  with_items:
+    - wget
+    - perl-Tk
+    - perl-Digest-MD5
+
+- name: texlive | download and extract installer
+  unarchive:
+    copy: no
+    creates: /tmp/install-tl-unx.tar.gz
+    dest: /tmp/
+    src: http://mirror.unl.edu/ctan/systems/texlive/tlnet/install-tl-unx.tar.gz
+
+- name: texlive | copy profile
+  copy: src=install-tl.profile dest=/tmp/install-tl.profile
+
+# Install path variables to the install-tl.profile file rather than
+# leaving them hard-coded in the profile file. This should be a foreach 
+# and a list, if such a thing is possible...
+- name: texlive | update profile file with install locations | TEXDIR
+  replace: 
+    regexp: '\$TEXDIR'
+    replace: "{{ TEXDIR }}"
+    dest: /tmp/install-tl.profile
+
+- name: texlive | update profile file with install locations | TEXMFLOCAL 
+  replace: 
+    regexp: '\$TEXMFLOCAL'
+    replace: "{{ TEXMFLOCAL }}"
+    dest: /tmp/install-tl.profile
+
+- name: texlive | update profile file with install locations | TEXMFSYSCONFIG
+  replace: 
+    regexp: '\$TEXMFSYSCONFIG'
+    replace: "{{ TEXMFSYSCONFIG}}"
+    dest: /tmp/install-tl.profile
+
+- name: texlive | update profile file with install locations | TEXMFSYSVAR 
+  replace: 
+    regexp: '\$TEXMFSYSVAR'
+    replace: "{{ TEXMFSYSVAR }}"
+    dest: /tmp/install-tl.profile
+
+- name: texlive | update profile file with install locations | TEXMFVAR
+  replace: 
+    regexp: '\$TEXMFVAR'
+    replace: "{{ TEXMFVAR }}"
+    dest: /tmp/install-tl.profile
+
+- name: texlive | install texlive
+  shell: rm /tmp/install-tl-unx.tar.gz
+  shell: cd /tmp/install-tl-* && ./install-tl -profile /tmp/install-tl.profile
+
+- name:  texlive | Symlink 
+  file: 
+    force: yes
+    path: /etc/fonts/conf.d/09-texlive.conf
+    src: "{{ TEXDIR }}/texmf-var/fonts/conf/texlive-fontconfig.conf"
+    state: link
+
+- name:  texlive | rebuild font cache
+  shell: fc-cache -rsfv
```

Comment:
Best practice, use [`command`](http://docs.ansible.com/ansible/command_module.html) instead of `shell` unless you need shell features